# Introduction

16S amplicon surveys have become a common technique for assessing bacterial community composition. The technique has been applied with great success towards understanding how bacterial communities and their individual members are associated various environments. While metagenomic approaches (shotgun sequencing of total nucleic acid extracts) has some major advantages over targeted amplicon sequencing, amplicon sequencing still offers a wealth of knowledge about how bacterial communities are structured at a fraction of the cost, and thus, amplicon based studies remain a practical option for researchers.

Analysis 16S amplicon survey data has been a standard analysis for over a decade. Surveys can be completed on sparse  (acid mine runoff) or dense (soil, human stool) communities. Regardless, the results of such experiments generate complex data sets composed of the information about the presence, relative abundances of and relationships of all detectable bacterial taxa within a community. These data are then associated with environmental factors (e.g. pH, soil depth, patient disease status) and a combination of both exploratory and statistical techniques are applied to determine if community structure or if a specific bacterial taxa is preferentially associated with an environment or state. This end-to-end workflow, beginning with sample collection and study design, proceeding through nucleic acid extraction, library preparation and sequencing requires a large number of decisions and planning by the researcher which ultimately results in a data file on their computer ready for analysis.

There are a number of well-developed software packages which enable researchers to analyze these frequently complex data sets. In particular, both [QIIME](http://qiime.org) [MOTHUR](https://www.mothur.org) stand out as applications that have enabled a tremendous amount of research into the investigation of 16S amplicon survey data. At the writing of this introduction, the original [QIIME publication](http://www.nature.com/nmeth/journal/v7/n5/full/nmeth.f.303.html) had been cited 7,154 times and the original [MOTHUR publication](http://aem.asm.org/content/75/23/7537.short) 6,551 times, quantifying the influence these software packages have had on the research community. 

While both of these software packages are "one-stop-shops" for 16S amplicon survey analysis, they do have some limitations. For example, many of the base graphics resulting from QIIME require post-processing steps in a graphic design program such as Adobe Illustrator or re-processing the data through R and plotting with ggplot2. Or if an analysis has not been developed by the QIIME or MOTHUR development teams then the researcher has to either wait for the analysis to be built into these packages, develop thier own software, or find another resource. This later challenge is being addressed by the plug-in capabilities of [QIIME 2](https://www.r-project.org), but there still remains a need for an integrated environment for managing 16S amplicon data, providing both exploratory and statistical analysis all while generating rich reports and plots to aid in the interpreation of results.

The [R programming language](https://www.r-project.org) provides a set of tools to, at least in part, complement analysis generated by these larger software packages. It is indeed possible to complete a study without any knowledge of R, but R does have an extensive history of development of algorithms for assessing community ecology (i.e. the [Vegan](https://cran.r-project.org/web/packages/vegan/vegan.pdf) software package original published in 2003) and testing for the differential abundance of taxa between states (i.e. [DeSeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html) and [metagenomeSeq](https://bioconductor.org/packages/release/bioc/html/metagenomeSeq.html)) that are native to R and may be of use to researchers in investigating their 16S amplicon data. In addition, R has extensive support for documenting workflows through the use of RMarkdown documents, an integrating native plotting environment and access to an advanced plotting paradigm through ggplot2, and perhaps most importantly, is useful for the analysis of other data types frequently created in research laboratories (RNAseq data, DNAmethylation, and basic statistical analysis). So not only does R provide some useful tools for the analysis of 16S amplicon data, the skills developed in using R for analysis of microbial communities could be applied to analysis of data from other types of studies.  For these reasons, R should be considered a valuable tool for researchers investigating bacterial communities.


## Motivation

## What this book is

The material in the book is focused heavily on the use of [PhyloSeq objects](https://joey711.github.io/phyloseq/import-data.html). PhyloSeq objects are a useful way to represent and integrate various types of data (e.g. operational taxonomic units or resolved sequence variant counts, taxonomy at various ranks, study variables and data, phylogenetic trees) into a single resource which can be accessed using functions in PhyloSeq, or extracted from to be used with other R packages (e.g. [ggplot2](http://ggplot2.tidyverse.org/reference/), [vegan](https://cran.r-project.org/web/packages/vegan/index.html)).

## What this book isn't