# Introduction

16S amplicon surveys have become a common technique for assessing bacterial community composition. The technique has been applied with great success towards understanding how bacterial communities and their individual members are associated various environments. While metagenomic approaches (shotgun sequencing of total nucleic acid extracts) has some major advantages over targeted amplicon sequencing, amplicon sequencing still offers a wealth of knowledge about how bacterial communities are structured at a fraction of the cost, and thus, amplicon based studies remain a practical option for many researchers.

16S amplicon survey's have been a standard way to study bacterial microbiomes for over a decade. Surveys can be completed on sparse  (acid mine runoff) or dense (soil, human stool) communities. Regardless, the results of such experiments generate complex data sets composed of the information about the presence and relative abundances of all detectable bacterial taxa within a community. These data are then associated with environmental factors (e.g. pH, soil depth, patient disease status) and a combination of both exploratory and statistical techniques are applied to determine if community structure or if the presence of specific bacterial taxa are preferentially associated with an environment or state. This end-to-end workflow, beginning with study design and sample collection, proceeding through nucleic acid extraction, library preparation and sequencing requires a large number of decisions and planning by the researcher which ultimately results in a data file on their computer ready for analysis.

There are a number of well-developed software packages which enable researchers to analyze these frequently complex data sets. In particular, both [QIIME](http://qiime.org) [MOTHUR](https://www.mothur.org) stand out as applications that have enabled a tremendous amount of research into the investigation of 16S amplicon survey data. At the writing of this introduction, the original [QIIME publication](http://www.nature.com/nmeth/journal/v7/n5/full/nmeth.f.303.html) had been cited 7,154 times and the original [MOTHUR publication](http://aem.asm.org/content/75/23/7537.short) 6,551 times, soundly establishing the influence these software packages have had on the research community. 

While both of these software packages are "one-stop-shops" for 16S amplicon survey analysis, they do have some limitations. For example, many of the base graphics resulting from QIIME require post-processing steps in a graphic design program such as Adobe Illustrator or re-processing the data through R. Similarly, if a particular analysis has not been developed by the QIIME or MOTHUR development teams then the researcher has to either wait for the analysis to be built into these packages, develop their own software, or find another resource. This later challenge is being addressed by the plug-in capabilities of [QIIME 2](https://www.r-project.org), but there still remains a need for an integrated environment for managing 16S amplicon data, providing both exploratory and statistical analysis all while generating rich reports and plots to aid in the interpretation of results.

The [R programming language](https://www.r-project.org) provides a set of tools to, at least in part, complement analysis generated by these larger software packages. It is indeed possible to complete a study without any knowledge of R, but R does have an extensive history of development of algorithms for assessing community ecology (i.e. the [Vegan](https://cran.r-project.org/web/packages/vegan/vegan.pdf) software package original published in 2003) and testing for the differential abundance of taxa between states (i.e. [DeSeq2](https://bioconductor.org/packages/release/bioc/html/DESeq2.html) and [metagenomeSeq](https://bioconductor.org/packages/release/bioc/html/metagenomeSeq.html)) and may be of use to researchers in investigating their 16S amplicon data. These R packages can be thought of as an App Store for microbiome analysis which has been contributed to for many decades from developers from a diversity of backgrounds. In addition to an extremely rich set of well-developed pacakges, R has 1) extensive support for documenting workflows through the use of R Markdown documents, 2) an integrating native plotting environment and access to an advanced plotting paradigm through ggplot2, 3) extensive reporting options including html and PDF reports and more sophistacted Dashboard reporting, and perhaps most importantly, 4) is useful for the analysis of other data types frequently created in research laboratories (RNAseq data, DNAmethylation, and basic statistical analysis). This is exemplified in the [Bioconductor](http://bioconductor.org) resource which conains hundreds of open-source pacakges designed to analzyze data from a variety of experiments. So not only does R provide some useful tools for the analysis of 16S amplicon data, the skills developed in using R for analysis of microbial communities could be applied to analysis of data from other types of studies.  For these reasons, R (a multitasker) should be considered a valuable tool for researchers investigating bacterial communities who seek a broader understanding of data anlysis beyond the confines of software packages limited to analyzing one type of data type (unitasker).

## Motivation

This book was written as a result of approximately 1 year of experience using a combination of QIIME and R to analyze 16S amplicon survey data generated to ask questions about what bacterial communities look like on different tissues of HIV infected patients. In addition, material had been produced for training [Workshops](http://evomics.org) over the past several years which made it apparent that a single software solution was not always sufficient for completion of a robust analysis. In the process of researching and training, it became apparent that there were a variety of resources (e.g. vignettes, forum posts). available to assist researchers in pursuing the daunting task of integrating software and algorithmic solutions to their analysis, but these were disparate and frequently challenging to find. Thus, this book represents an effort to synthesize information from these experiences and resources into a cohesive unit with the hopes that others may benefit. 

## What this book is

One general misconception about microbiome analysis that there is a singularity within each analysis platform. So people will state that they used QIIME, MOTHUR or PhyloSeq to analyze their data. In reality, this is misleading as each of these offer distinct capabilities and integrate various programs, concepts and algorithms from a variety of sources. So each is a framework to be utilized for very specific tasks, not to just "do microbiome analysis". For example, instead of stating that one used QIIME to anlyze their data, it would be more accurate and more useful to other researchers to know 1) how the raw data were QC'd, using QIIME or another program. If in QIIME what settings were use, 2) how the data were demultiplexed and quality controlled, 2) 
 
 The material in the book is focused heavily on the use of [PhyloSeq objects](https://joey711.github.io/phyloseq/import-data.html).

PhyloSeq objects are a useful way to represent and integrate various types of data (e.g. operational taxonomic units or resolved sequence variant counts, taxonomy at various ranks, study variables and data, phylogenetic trees) into a single resource which can be accessed using functions in PhyloSeq, or extracted from to be used with other R packages (e.g. [ggplot2](http://ggplot2.tidyverse.org/reference/), [vegan](https://cran.r-project.org/web/packages/vegan/index.html)). The content is focused on data curation and manipulation, workflow design, analysis tools and plotting not only using the built-in functions within PhyloSeq, but also on pulling data from PhyloSeq objects for use in other packages.

## What this book isn't

This book is not meant to explain why you should use R over QIIME or MOTHUR or a variety of other options that are available for 16S amplicon survey analysis. Quite contrary to this, it is focused on how R can be used to integrate analysis from a variety of sources and output them to an even wider variety of resources. So the content is meant to be complementary to what is offered in these well-developed software packages.

This book is also not designed to replace the well-developed [PhyloSeq vignettes](https://joey711.github.io/phyloseq/) or other tutorials found around the web. In fact, much of this material was synthesized after going through these same vignettes and tutorials for various research projects and will pull directly from parts of them. Similarly, while it provides some basic material on how to work within R in the chapter on General Skills, it will not be enough material to make you an "R Wizard". There is vast amounts of material on the web or through University courses that will assist in advanced training in R if you so desire. 

Finally, this book is not designed to provide in-depth explanations of concepts in microbial ecology. That topic requires several books on its own. However, basic introductions to some core concepts are provided for explanatory purposes.

While this book does not formally cover these concepts, there is an appendix which provides links to various resources used in the creation of the content of this book, or which one may find useful for expanding their own knowledge base.