---
title: "Microbiome Analysis Using R"
author: "Scott A. Handley"
date: "2017-04-13"
site: bookdown::bookdown_site
documentclass: book
bibliography: [book.bib]
biblio-style: apalike
link-citations: yes
github-repo: shandley/microbiome_analysis_using.R
description: "Microbiome analysis using R."
---

# Preface {-}

This book was designed to provide instruction and examples on how to process 16S amplicon data in R for microbial community ecology studies. The material was developed during over a period from Fall 2016 through Summer 2017 and represents a sythesis of material obtained during training at the [2017 Workshop on Genomics](http://evomics.org) and through forum posts across the internet. In particular, the [PhyloSeq](https://joey711.github.io/phyloseq/) and [dada2](https://github.com/benjjneb/dada2) GitHub issues page.

The material in the book is focused heavily on the use of [PhyloSeq objects](https://joey711.github.io/phyloseq/import-data.html). PhyloSeq objects are a useful way to represent and integrate various types of data (e.g. operational taxonomic units or resolved sequence variant counts, taxonomy at various ranks, study variables and data, phylogenetic trees) into a single resource which can be accessed using functions in PhyloSeq, or extracted from to be used with other R packages (e.g. [ggplot2](http://ggplot2.tidyverse.org/reference/), [vegan](https://cran.r-project.org/web/packages/vegan/index.html)).